src/js/stores/font.js

```diff
@@ -138,27 +138,41 @@ define(function (require, exports, module) {
                 psName = textStyle.postScriptName,
                 fontObj = this._postScriptMap.get(psName, null);
                 
-            if (!fontObj) {
-                // FIXME: What about missing fonts?
-                throw new Error("The font for layer is not available!");
-            }
+            // If it's mixed, or there is no fontObj, we skip adding these values to the type object
+            // adbeFont, fontFamily, fontStyle, fontWeight
+            if (fontObj) {
+                var family = fontObj.family,
+                    fontRecord = this._familyMap.get(family),
+                    psObj = fontRecord.get(fontObj.font, null);
+
+                if (!psObj) {
+                    throw new Error("The font for layer is not available!");
```

Comment:
I'm curious as to why you chose to throw here. This will cause `duplicateTextStyle` to reject. Is that what you want? Would it be better to just log a warning, return null, and then make sure callers handle that scenario?